cmd/juju/action/run.go

```diff
@@ -241,18 +255,30 @@ func (c *runCommand) Run(ctx *cmd.Context) error {
 		return errors.Errorf("params must be a map, got %T", typedConformantParams)
 	}
 
-	actions := make([]params.Action, len(c.unitTags))
-	for i, unitTag := range c.unitTags {
-		actions[i].Receiver = unitTag.String()
+	actions := make([]params.Action, len(c.unitReceivers))
+	for i, unitReceiver := range c.unitReceivers {
+		if strings.HasSuffix(unitReceiver, "leader") {
```

Comment:
does this fail for "application-leader"  I don't think it does, but I figured I'd check. I'm also curious why you're just using 'suffix' rather than the regex you defined earlier. (i would guess we don't actually need the regex).

Reply:
The regular expression is used in _Init_ to validate all the arguments; supplying "application-leader" does indeed fail.

By the time we get down here, we know we have valid unit designators. I think it is OK to check the suffix, which will be less overhead than the regex (not that we care much here).